packages/fast-colors/src/component-state-color-palette.ts

```diff
@@ -0,0 +1,165 @@
+import { ColorRGBA64 } from "./color-rgba-64";
+import { ColorPalette } from "./color-palette";
+import { hslToRGB, rgbToHSL, rgbToLuminanceLinear } from "./color-converters";
+import { ColorScale, ColorScaleStop } from "./color-scale";
+import { ColorHSL } from "./color-hsl";
+import { ColorInterpolationSpace } from "./color-interpolation";
+import { parseColorHexRGB } from "./parse-color";
+
+export interface ComponentStateColorPaletteConfig {
+    baseColor?: ColorRGBA64;
+    steps?: number;
+}
+
+export class ComponentStateColorPalette {
+    public static readonly defaultPaletteConfig: ComponentStateColorPaletteConfig = {
+        baseColor: parseColorHexRGB("#808080"),
+        steps: 94,
+    };
+
+    public palette: ColorRGBA64[] = [];
+    private readonly config: ComponentStateColorPaletteConfig;
+
+    constructor(config: ComponentStateColorPaletteConfig) {
+        this.config = Object.assign(
+            {},
+            ComponentStateColorPalette.defaultPaletteConfig,
+            config
+        );
+        this.palette = [];
```

Comment:
`this.palette = []` is unnecessary - you're already defined it as an empty array with `public palette: ColorRGBA64[] = []`